daemon: Don't auto disable session affinity (#16179)
cilium-team/cilium: Documentation/gettingstarted/kubeproxy-free.rst

The source for the affinity depends on the origin of a request. If a request is
sent from outside the cluster to the service, the request's source IP address is
used for determining the endpoint affinity. If a request is sent from inside
the cluster, then the source depends on whether the :ref:`host-services` feature
is used to load balance ClusterIP services. If yes, then the client's network
namespace cookie is used as the source. The latter was introduced in the 5.7
Linux kernel to implement the affinity at the socket layer at which
:ref:`host-services` operate (a source IP is not available there, as the
endpoint selection happens before a network packet has been built by the

> kernel). If :ref:`host-services` is not used (i.e. the loadbalancing is done by
> the BPF program ``bpf_lxc`` attached to a pod network interface), then the

Reviewer: Maybe we can avoid references to BPF program names in a documentation destined to users?
```suggestion
kernel). If :ref:`host-services` is not used (i.e. the loadbalancing is done at
the pod network interface, on a per-packet basis), then the
```

select the same service endpoint for a given service with session affinity configured.
To disable the feature, set ``config.sessionAffinity=false``.

For users who run with kube-proxy (i.e. with Cilium's kube-proxy replacement
disabled), the ClusterIP service loadbalancing when a request is sent from a pod

> running in a non-host network namespace is still performed by the ``bpf_lxc``

Reviewer: Ditto.
```suggestion
running in a non-host network namespace is still performed at the pod network interface
```

(until `GH#16197<https://github.com/cilium/cilium/issues/16197>`__ is fixed).
For this case the session affinity support is disabled by default. To enable the
feature, set ``config.sessionAffinity=true``.

> 

Reviewer: I think we also need to document the kernel requirement in https://docs.cilium.io/en/latest/operations/system_requirements/.
